### ISSUE TITLE
Any Guide for integrating more SOTA models?

### ISSUE THREAD

**Issue by ShivamSRS** [bytedance/LVLM_Interpretation#8]
Hi authors

I am trying to use this repo for an array of models like MedGemma, Gemma3, etc. Is there a guide to what are the integration endpoints one needs to be aware of, that the authors can offer?

Thanks for your great work!

**Comment by xiaoying-xing** [comment]
Hi, thanks for you interest in this work. You will need to run the model inference correctly (such as chat templates, preprocessing), apply the mask to the input image, and get their outputs and logits for the mask optimization. In the code we have provided examples of different models, you can find some if-- else-- with model names.

**Comment by ShivamSRS** [comment]
Hi @xiaoying-xing 
Thank you so much for your response! That is encouraging to hear. So it is not required that the user records/changes the internals of the model architecture or add hooks of any kind to any internal layers? The entire process can work with the model as a black box, as long as the logits and outputs are provided?

**Comment by xiaoying-xing** [comment]
No, you don't need to add any hooks into the internal layers.
